src/NServiceBus.Core/UnitOfWork/IManageUnitsOfWork.cs

```diff
@@ -7,6 +7,7 @@ namespace NServiceBus.UnitOfWork
     /// Interface used by NServiceBus to manage units of work as a part of the
     /// message processing pipeline.
     /// </summary>
+    [ObsoleteEx(ReplacementTypeOrMember = "Behavior", TreatAsErrorFromVersion = "9", RemoveInVersion = "10")]
```

Comment:
@andreasohlund @DavidBoike I wasn't really sure what to write for `ReplacementTypeOrMember` and whether we should also provide a `Message`.

Same applies to the other two types.

Reply:
Just gets concatenated into the generated Obsolete attribute. What about this?
```suggestion
    [ObsoleteEx(Message = "The unit of work pattern is more straightforward to implement in a pipeline behavior, where the using keyword and try/catch blocks can be used.", ReplacementTypeOrMember = "Behavior<TContext>", TreatAsErrorFromVersion = "9", RemoveInVersion = "10")]
```